src/components/views/messages/DecryptionFailureBody.tsx

```diff
@@ -27,6 +27,12 @@ function getErrorMessage(mxEvent?: MatrixEvent): string {
 }
 
 // A placeholder element for messages that could not be decrypted
```

Comment:
could you document the new behaviour please. "Accepts a property `ref` , which, if provided, will be set to the DOM node of the rendered element."

Reply:
I disagree. The types allow a `ref` property but don't tell you what it will be set to.

This is my whole "you shouldn't have to read the code of a function to know how to use it" thing.

Reply:
I'm not suggesting reading the code, just the type you import.

![image](https://user-images.githubusercontent.com/2403652/234585287-e6d38b5f-6056-43fe-8d55-e19ff34e43d1.png)

It says accepts a ref that is passed to a HtmlDivElement

Reply:
I feel like decoding that type is pretty advanced, and it's certainly not obvious to me as a casual reader of the code. Honestly, the type - even if I manage to get my IDE to tell me what it is - opens more questions than it answers, starting with 'WTF is a "ForwardRefExoticComponent"?' 'Exotic' sounds like something complicated I don't need to know about. And understanding the nesting of those `< ... >` parameters? That's not a human-readable type IMHO.

I'm not going to insist on this (consider me -0.7 on the [Apache voting scale](https://www.apache.org/foundation/voting.html#expressing-votes-1-0-1-and-fractions) for landing this without more comments), but I think spelling this sort of thing out makes our code a lot more accessible.

Reply:
> It says accepts a ref that is passed to a HtmlDivElement

Also, assuming you're talking about `RefAttributes<HtmlDivElement>`, I don't think this is correct? It accepts a ref that *will be populated with* an `HtmlDivElement` (though that part of the type alone doesn't indicate who is responsible for doing that population).

Reply:
Sure but as the caller that's not something you should care about, that's an implementation detail. Whether we manually assign `ref.current` in the component or we trust React Fibre to do it because we pass it to `ref` on a built in has 0 discernible difference to the caller.

Reply:
The `ref` prop name in React is special cased. You cannot have a normal prop called `ref` and it work other than how React wants it to work. The thing you're passing to, either directly or indirectly will set the Ref.

Reply:
(taken out of this thread)